### PR TITLE
[DEV-4050] align goals query

### PR DESCRIPTION
**Pull request by gaetano-guerriero**

### CODE REVIEW

File: server/athenian/api/align/queries/teams.py [hunk 1]
```diff
@@ -68,21 +68,33 @@ def build_team_tree_from_rows(team_rows: Iterable[Mapping[str, Any]],
...
     team_rows = await fetch_teams_recursively(
         account, sdb, team_select, [root_team_id] if root_team_id else None,
     )
-    nodes, root_team_id = build_team_tree_from_rows(team_rows, root_team_id)
-    return _build_team_tree(nodes[root_team_id], nodes)
+    return build_team_tree_from_rows(team_rows, root_team_id)
 
 
-def _build_team_tree(team_info: Dict[str, Any], all_teams: dict) -> TeamTree:
+def build_team_tree_from_rows(rows: Sequence[Row], root_team_id: Optional[int]) -> TeamTree:
```
Comment (gaetano-guerriero) [inline server/athenian/api/align/queries/teams.py:82]: fixed nomenclature among fetch/build and tree/nodes, and added this new entrypoint

File: server/athenian/api/align/queries/goals.py [hunk 2]
```diff
@@ -1,15 +1,135 @@
...
+        group_teams = {
+            row[TeamGoal.team_id.name]: team_member_map[row[TeamGoal.team_id.name]]
+            for row in group_team_goal_rows
+        }
+        group_goal_row = group_team_goal_rows[0]  # could be any row
+        metric = TEMPLATES_COLLECTION[group_goal_row[Goal.template_id.name]]["metric"]
+
+        calc_metrics = partial(
+            calculate_team_metrics,
+            metrics=[metric],
```
Comment (vmarkovtsev) [inline server/athenian/api/align/queries/goals.py:63]: Performance: we should calculate all the metrics at once.

Comment (vmarkovtsev) [reply server/athenian/api/align/queries/goals.py:63]: Both initial and current. The underlying calculators accept multiple time intervals.

Comment (vmarkovtsev) [reply server/athenian/api/align/queries/goals.py:63]: `calculate_team_metrics` should be extended to accept several time intervals, and to return the equivalent number of metric value trees.

**Review by vmarkovtsev — CHANGES_REQUESTED** [review]

File: server/athenian/api/align/queries/goals.py (hunk 2 above)
Comment (gaetano-guerriero) [reply server/athenian/api/align/queries/goals.py]: now it's really done

**Review by vmarkovtsev — APPROVED** [review]

**Comment by vmarkovtsev** [comment]
Thanks!

**Pull request merged**